### PR TITLE
Soa bot/dipnet daide parser

### LINKED ISSUE #45: Convert Dipnet orders to DAIDE
See https://github.com/trigaten/DAIDE/issues/3

We need a function to be placed in `utils.py` that can do this.

### PR DESCRIPTION
**Pull request by kartik2112**
Added translator between dipnet and DAIDE to utils.py. Currently supports all types of orders.

Convoy orders translation scoped to the assumption that the army move orders through sea and the convoying fleets are in the set of orders by the same power. If it isn't, this function might not work appropriately.

Resolves #45

### CODE REVIEW

**Comment by trigaten** [comment]
CI is failing

**Review by trigaten — CHANGES_REQUESTED** [review]
Nice docstrings and type hints! Can you add more "#" comments throughout the code?

File: src/baseline_bots/utils.py [hunk 1]
```diff
@@ -236,7 +236,177 @@ def get_province_from_order(order):
...
+
+    # For each order
+    for dipnet_order_tokens in dipnet_style_order_strs_tokens:
+
+        # Create unit to power mapping for constructing DAIDE tokens
+        unit_game_mapping = {}
+        for power in list(game.powers.keys()):
+            for unit in game.get_units(power):
+                unit_game_mapping[unit] = power[:3]
+        # def bfs_fleets(src, dest, game):
```
Comment (trigaten) [inline src/baseline_bots/utils.py:289]: remove this?

File: src/baseline_bots/utils.py [hunk 2]
```diff
@@ -236,7 +236,177 @@ def get_province_from_order(order):
...
+            raise f"error from utils.dipnet_to_daide_parsing: unit {dipnet_suborder} not present in unit_game_mapping"
+        return "(" + (" ".join(
+            [
+                unit_game_mapping[dipnet_suborder],
+                "AMY" if dipnet_suborder[0] == "A" else "FLT",
+                dipnet_suborder.split()[-1]
+            ]
+        ) ) + ")"
+    
+    if type(dipnet_style_order_strs) != list:
```
Comment (trigaten) [inline src/baseline_bots/utils.py:267]: Why does this need to be checked?

Comment (kartik2112) [reply src/baseline_bots/utils.py:267]: Will remove.

File: src/baseline_bots/utils.py [hunk 3]
```diff
@@ -236,7 +236,172 @@ def get_province_from_order(order):
     else:
         return order_tokens[0]
 
+def dipnet_to_daide_parsing(dipnet_style_order_strs: List[str], game: Game) -> List[str]:
+    """
+    Convert dipnet style single order to DAIDE style order. Needs game instance to determine the powers owning the units
+    * dipnet_style_order_strs: dipnet style list of orders to be converted to DAIDE
```
Comment (trigaten) [inline src/baseline_bots/utils.py:242]: This is not a huge deal, but lets try to do sphinx style docstrings,

so :param dipnet_style_order_strs: dipnet style list of orders to be converted to DAIDE

:return: ..., etc.

**Review by trigaten — APPROVED** [review]
Nice work! The code looks good :)

**Pull request merged**